resharper/resharper-unity/src/UnityFindReferencedCodeProvider.cs

```diff
@@ -0,0 +1,50 @@
+using JetBrains.Application;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Feature.Services.Navigation.Requests;
+using JetBrains.ReSharper.Features.Navigation.Features.FindUsages;
+using JetBrains.ReSharper.Psi;
+
+namespace JetBrains.ReSharper.Plugins.Unity
+{
+    [ShellComponent]
+    public class UnityFindUsagesProviderProvider : FindUsagesProvider
```

Comment:
Too many `Provider`s :) Change name to `UnityFindUsagesProvider`.

Also, move this to `CSharp\Feature\Navigation\FindUsages`

I'm not really keen on this class. I don't like deriving from a core component. We're a plugin, and if other plugins tried to do this it would break. Perhaps speak to the team and see if we can have a proper extension point?

Reply:
I will rollback changes and talk about extension point

Reply:
Don't worry about rolling it back right now, but do ask.